src/core/common/Buttons/CustomShowButton.js

```diff
@@ -0,0 +1,25 @@
+import React from "react";
+import { ShowButton } from "react-admin";
+
+import { withStyles } from '@material-ui/core/styles';
+
+const styles = {
+    showButton: {
+        color: '#30ad57',
+        fontSize: 16,
```

Comment:
I don't understand, why, but ShowButton doesn't work with theme settings (if we compare with CreateButton):

```
// This styles will be used in ShowButton
const styles = {
    showButton: {
        color: '#30ad57',
        fontSize: 16,
        textTransform: 'capitalize',
    },
};

// In this case ShowButton doesn't use custom styles, and uses default view
const styles = theme => ({
    showButton: {
        color: theme.palette.completeButton,
        fontSize: theme.buttonFontSize,
        textTransform: 'capitalize',
    },
});
```

I tried to solve this issue by custon ShowButton creating: 
```
const styles = theme => ({
....
})

const CustomShowButton = ({ classes, ...rest }) => (
    <Button className={classes.showButton} onClick={() => someFunction()}>
            View
    </Button>
);
```
But in this case clicking on the ShowButton occurs redirecting to PatientSummary page (because we click on Datagrid row) instead of redirecting to the Show page.

Reply:
There is no reason why the showButton can't work with a theme. You should dig further. Are the theme properties that you're using (`palette.completeButton` and `buttonFontSize` defined in the theme?

Reply:
@fzaninotto 
Yes, **completeButton** and **buttonFontSize** were defined, when I try to realize this solution. 
But even if we will use contants styles with **theme** don't working for ShowButton:
```
const styles = theme => ({
    showButton: {
        color: '#30ad57',
        fontSize: 16,
        textTransform: 'capitalize',
    },
});
```
In this case button will have default view:
![image](https://user-images.githubusercontent.com/39263134/54294357-0e708700-45ba-11e9-906a-2a03835bec70.png)

Reply:
That's very weird. Can you check the `classes` injected into the component? What are the generated styles for these classes in the DOM?

Reply:
Yes, I checked - classes property is equal in both cases:

- with themes
![002](https://user-images.githubusercontent.com/39263134/54340447-0bb87500-4640-11e9-9915-ed98a6d3a93f.png)

- without themes
![001](https://user-images.githubusercontent.com/39263134/54340458-13781980-4640-11e9-8ee7-35421c17eeb3.png)

Reply:
then it means the problem isn't in the generation of class names, but in the generation of styles. Can you check with the Chrome debug tool which styles the class name contains in each case?

Reply:
I assumed, that the reason in className and renamed it to "myShowButton" - only for testing.

So, in case without **theme** we can see, that styles for the element **CustomShowButton-myShowButton-271** are presented:
![0003](https://user-images.githubusercontent.com/39263134/54351534-5940dc00-4658-11e9-972a-669c8ad3a657.png)

In opposite case styles for the element **CustomShowButton-myShowButton-271** are absent generally (I scrolled right area to the bottom - nothing)
![0004](https://user-images.githubusercontent.com/39263134/54351700-bb99dc80-4658-11e9-937e-33adba4d5a5c.png)

Only one difference between CreateButton and ShowButton. First is a button, but the second one is a link.

Reply:
@fzaninotto This issue isn't actual more. 
@tony-shannon said to change behavior and function of this button:
  
Click on Row to open details panel
Click on View to redirect to summary please

I realized it and used color settings from theme.

Buttons sizes from theme will be done in the next PR, for styling changes.